### PR TITLE
Feature/grab bag

### PR DESCRIPTION
**Pull request by denis-yuen**
* add github actions for validating cwl and running the workflows
* clean-up json schema to eliminate warnings
* adding into github apps (eating own dogfood)

Think this is what we can do on our end for https://github.com/dockstore/dockstore/issues/4886 
Ultimately, it is valid CWL, so I think the rest will need to be tackled on the SBG side (they shouldn't reject valid CWL)

### CODE REVIEW

**Comment by denis-yuen** [comment]
FYI, invited @aofarrel to this organization and for review

**Review by coverbeck — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

**Review by aofarrel — CHANGES_REQUESTED** [review]
I'm coming at this from the perspective of making this as simple as possible while also being fully tested. I might be missing some context here, but I have a few concerns with this PR as-is.

* It looks like the nextflow version is not getting tested by GitHub actions. Is that something we can implement?
* cli-script.sh seems unnecessary; I get that it's getting called by workflows.yml but the WDL equivalent doesn't call a script at all. imho we should either have a .sh file for all workflow runs, or neither. If there's a technical reason for the CWLs getting their own script file then I think cli-script.sh should be renamed cli-cwl-test.sh for clarity's sake.
* Validate all the things does not validate all the things; see comments.

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        python-version: '3.9'
+
+    - name: Setup engines
+      run: | 
+        curl -o requirements.txt "https://raw.githubusercontent.com/dockstore/dockstore/1.13.0-beta.3/dockstore-webservice/src/main/resources/requirements/1.13.0/requirements3.txt"
+        pip3 install --user -r requirements.txt
+
+    - run: sudo apt-get install -y moreutils
+
+    - name: Validate all the things
```
Comment (aofarrel) [inline .github/workflows/validate.yml]: This only validates the CWL. Consider adding the WDL and NFL equivalents too. Not sure about nextflow but for WDL the command would be `java -jar womtool.jar validate [whatever.wdl]` which will return 0 plus the text "Success!" if it passes.

Comment (aofarrel) [reply .github/workflows/validate.yml]: Strictly speaking validating the WDL isn't fully necessary -- when `dockstore --script  workflow launch --local-entry checker.wdl  --json md5sum.wdl.json` runs, validation will occur first. But if we go to the trouble of validating the CWL, why not go to that trouble for everything?

File: .github/workflows/workflows.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        printf "cromwell-vm-options: -DLOG_LEVEL=ERROR\nserver-url: https://dockstore.org/api" > ~/.dockstore/config
+        dockstore
+
+    - name: Run cwl workflows with CLI
+      run: |
+        bash cli-script.sh
+
+    - name: Run wdl workflow with CLI
+      run: |
+        dockstore --script  workflow launch --local-entry checker.wdl  --json md5sum.wdl.json
```
Comment (aofarrel) [inline .github/workflows/workflows.yml:44]: Regardless of whether or not cli-script.sh should exist or not, "Run cwl workflows with CLI" as is runs `set -eux -o pipefail` before running the Dockstore CLI, but my understanding "Run wdl workflow with CLI" 's Dockstore call does not operate under these options. Whether we keep cli-script.sh or not, I think `set -eux -o pipefail` needs to be added to the WDL context.

File: .github/workflows/validate.yml (hunk 1 above)
Comment (denis-yuen) [reply .github/workflows/validate.yml]: Adding wdl validation

File: .github/workflows/workflows.yml (hunk 2 above)
Comment (denis-yuen) [reply .github/workflows/workflows.yml:44]: Removing script to be consistent

**Comment by denis-yuen** [comment]
> It looks like the nextflow version is not getting tested by GitHub actions. Is that something we can implement?

I don't think there is a nextflow version in this repo

**Comment by aofarrel** [comment]
> > It looks like the nextflow version is not getting tested by GitHub actions. Is that something we can implement?
> 
> 
> 
> I don't think there is a nextflow version in this repo

Sorry, I meant WDL. Need a stronger brew... 🫖

**Comment by aofarrel** [comment]
LGTM!

(This may go through as a comment rather than a formal approval - I'm OOO so I'm using a new GH client. Feel free to merge as-is.)

**Pull request merged**